### ISSUE TITLE
help command and extension docs

### ISSUE THREAD

**Issue by bpoldrack** [datalad/datalad#5526]
Idea came about in https://github.com/datalad/datalad-metalad/issues/86.

For documentation we basically have the `--help` option of commands only. For some aspects we need a way to have conceptual documentation, that isn't tied to a particular command. In the issue above it is about some general concepts of metalad, but it would be useful for other extensions and some things in core, too.

So: Have a `help` command, that uses some extension "entrypoint" to provide a general help page for extensions. It should be possible to also allow for generating arbitrary topic pages. Like `datalad help extension-name`, but also `datalad help ria-store`, `datalad help ria-url`, `datalad help archives-special-remote`, etc.

Ideally respective man pages would be rendered from that, too `man extension-name`, `man datalad-ria-store`, etc.

What do you think, @datalad/developers?

**Comment by kyleam** [comment]
I think such a command sounds nice, but I'm not sure it's worth the extra complexity (more entry points and datalad-specific customization as well as one more way to view documentation).  For these extension overviews and conceptual pages, wouldn't `pydoc` be sufficient?  For extensions, write the top-level `__init__`.  For concepts, a new file (restricted to the standard python module naming) could be added if there isn't a fitting module already.

Command help pages could then point to the relevant topic pages, spelling out the appropriate `pydoc` invocation.  `datalad --help` could point to the extension overviews.

**Comment by yarikoptic** [comment]
Makes total sense.  Minor concern: since commands themselves have `--help` it would suggest having two ways for seeking help which might add a bit of confusion: when do I use `help` vs `--help`, which should be mentioned in output of `datalad help extension` to seek detailed help for individual command.

Also I guess plain `datalad help`, similarly to `datalad --help` listing all commands, should list all extensions.  

Following @kyleam 's comment -- top-level `__doc__` of extensions could provide the description and list of entry points for an extension would be compiled automagically

**Comment by yarikoptic** [comment]
another note: I still hope that at some point we would reincarnate attempt for integration with helpme #4586 to automate querying (absence of that stopped us somewhat) and submission of issues upon crashes etc. Not sure if `datalad helpme` would make sense though in that context.

**Comment by kyleam** [comment]
> Well, I'd argue `pydoc`, too, would be "one more way" given that we nowhere point to it already

I'd very much consider `pydoc` an existing approach rather than one more way.  It is the built-in tool for viewing python documentation, and it is what `help` uses underneath in the python console.  I don't think every python library/package with docstrings needs to point to it for it to count as a way to view docs.

**Comment by bpoldrack** [comment]
> I don't think every python library/package with docstrings needs to point to it for it to count as a way to view docs.

That's true, but we have command line only users, who wouldn't know anything about python packages. If we go that way (only), we need to point them to it.

However, I suppose we can have both. First, put those docs in respective modules and thereby make them available via pydoc.
If add some kind of registration of relevant modules, we can then also generate manpages and have a cheap `help` command. I'll admit that the second step is more in the category of nice-to-have, though.

**Comment by kyleam** [comment]
> Do we loose "special rendering" (i.e. `[PY]` vs `[CMD]`)?

No, it's not lost; you get the python rendering. (`python -m pydoc ...` is actually the way that I check `[PY]` rendering when I'm developing command docs.)

> How easily could we generate man pages from those places w/o generating a page for each and every module?

I don't know.  I was proposing that you just rely on `pydoc` and not worry about generating manpages.


> If we go that way (only), we need to point them to it.

Right, as I mentioned above, I think the `pydoc` invocation should be spelled out.

Anyway, I've contributed my two cents :)
